otx/algorithms/segmentation/adapters/mmseg/task.py

```diff
@@ -0,0 +1,563 @@
+                    labels=self._labels,
+                )
+
+        self._is_training = True
+
+        self._init_task()
+
+        cfg = self.configure(True, "train", None)
+        logger.info("train!")
+
+        timestamp = time.strftime("%Y%m%d_%H%M%S", time.localtime())
+
+        # Environment
+        logger.info(f"cfg.gpu_ids = {cfg.gpu_ids}, distributed = {cfg.distributed}")
+        env_info_dict = collect_env()
+        env_info = "\n".join([(f"{k}: {v}") for k, v in env_info_dict.items()])
+        dash_line = "-" * 60 + "\n"
+        logger.info(f"Environment info:\n{dash_line}{env_info}\n{dash_line}")
+
+        # Data
+        datasets = [build_dataset(cfg.data.train)]
+
+        # FIXME: Currently segmentor does not support multi batch evaluation.
+        # For the Self-SL case, there is no val data. So, need to check the
+
+        if "val" in cfg.data and "val_dataloader" in cfg.data:
+            cfg.data.val_dataloader["samples_per_gpu"] = 1
+
+        # Target classes
+        if "task_adapt" in cfg:
+            target_classes = cfg.task_adapt.final
+        else:
+            target_classes = datasets[0].CLASSES
+
+        # Metadata
+        meta = dict()
+        meta["env_info"] = env_info
+        meta["seed"] = cfg.seed
+        meta["exp_name"] = cfg.work_dir
+        if cfg.checkpoint_config is not None:
+            cfg.checkpoint_config.meta = dict(
+                mmseg_version=__version__ + get_git_hash()[:7],
+                CLASSES=target_classes,
+            )
+
+        # Model
+        model = self.build_model(cfg, fp16=cfg.get("fp16", False))
+        model.train()
+        model.CLASSES = target_classes
+
+        if cfg.distributed:
+            torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
+            if cfg.dist_params.get("linear_scale_lr", False):
+                new_lr = len(cfg.gpu_ids) * cfg.optimizer.lr
+                logger.info(
+                    f"enabled linear scaling rule to the learning rate. \
+                    changed LR from {cfg.optimizer.lr} to {new_lr}"
+                )
+                cfg.optimizer.lr = new_lr
+
+        validate = bool(cfg.data.get("val", None))
+        train_segmentor(
+            model,
+            datasets,
+            cfg,
+            distributed=cfg.distributed,
+            validate=validate,
+            timestamp=timestamp,
+            meta=meta,
+        )
+
+        # Save outputs
+        output_ckpt_path = os.path.join(cfg.work_dir, "latest.pth")
+        best_ckpt_path = glob.glob(os.path.join(cfg.work_dir, "best_mDice_*.pth"))
+        if len(best_ckpt_path) > 0:
+            output_ckpt_path = best_ckpt_path[0]
+        best_ckpt_path = glob.glob(os.path.join(cfg.work_dir, "best_mIoU_*.pth"))
+        if len(best_ckpt_path) > 0:
+            output_ckpt_path = best_ckpt_path[0]
+        return dict(
+            final_ckpt=output_ckpt_path,
+        )
+
+    def _get_feature_module(self, model):
```

Comment:
It seems this function is not used, and other algo do not contain this function. Please remove

Reply:
Fixed, please review again, thanks!